lib/entice/logic/skills/skill.ex

```diff
@@ -92,8 +96,31 @@ defmodule Entice.Logic.Skill.Behaviour do
   defcallback apply_effect(target_entity_id :: term, caster_entity :: %Entity{}) ::
     :ok |
     {:error, reason :: term}
+
+  @doc "Is called before starting to cast."
+  defcallback prerequisites_fulfilled?(target_entity_id :: term, caster_entity :: %Entity{}) ::
```

Comment:
the questionmark syntax is actually reserved for functions returning a boolean... the other module uses it for other functions which is suboptimal, but we can fix that at some other point. though i think with this macro stuff we wanna get it right the first time we implement it - i'd vote therefor for an active short version... maybe 'check_requirements'?
also, we can indicate that we want to accept both ids and entities for the target like so:
```elixir
...(target_entity_id_or_entity :: term | %Entity{}, ...
```